### DISCUSSION TITLE
hexadecimal fields in CSV file

### DISCUSSION THREAD

**Discussion by tloubrieu-jpl** [NASA-PDS/validate#510] [Q&A] [answered]
Hello,

With the attached files [Archive.zip](https://github.com/NASA-PDS/validate/files/8825432/Archive.zip), I am getting the following errors from validate version 2.2.0:

    ./bin/validate -t <folder where the files are> --skip-context-validation

      ....
      ERROR  [error.table.field_value_format_specifier_mismatch]   table 1, record 2, field 5: The value '600c' does not match the defined field format specifier 'x': For input string: "600c"
     ...

I don't see how I should write these hexadecimal field values otherwise, could thast be a bug in validate ?

Thanks

Thomas

**Comment by rchenatjpl** [comment]
this is not the problem, but your .xml has two field_number = 6. I'm still looking at the rest

**Comment by tloubrieu-jpl** [comment]
Thanks @rchenatjpl I will update the redundant field number

**Comment by rchenatjpl** [comment]
Attached are Thomas's files modified:
- the .CSV is chopped to 6 fields so that all data lines are the same length
- stuff irrelevant to this problem is stripped from the .xml
- the Table_Delimited is copied to a Table_Character and modified appropriately. 'diff *.xml' to see
So:
- validate incorrectly flags valid hex values, as Thomas originally pointed out
- validate seemingly ignores field_format for Table_Character but considers it for Table_Delimited
- validate considers data_type for both. If you change a value to something non-hex, an extra error is reported for both
Those are the facts. More is just opinion, and I'm confident you guys know what to do.

There was more little stuff irrelevant to this issue that you would have eventually caught:
- The last dash in title (before "ICO Unknown Activity") was a non-ASCII dash
- No blank space (specifically carriage-returns and spaces) within logical_identifier or lid_reference
- No caps in LIDs either
So (the carriage-return isn't visually showing correctly):
`        <logical_identifier>
           urn:nasa:pds:clipper.ecm:data-register-raw:XXX000XXX_...</...>
should be
        <logical_identifier>urn:nasa:pds:clipper.ecm:data-register-raw:xxx000xxx_...</...>
`

[Archive.zip](https://github.com/NASA-PDS/validate/files/8827612/Archive.zip)

**Comment by jordanpadams** [comment, marked as answer]
Created new issue to track fix this https://github.com/NASA-PDS/validate/issues/511
